src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala

```diff
@@ -16,34 +16,44 @@ class TestVariableReaderLabelBased extends Test {
   case class VariableTest(
      name: String, text: String,
      // If there are multiple "Assignment" mentions, use one variable for each and multiple lines.
+     label: String,
      variables: Seq[Variable]
    ) {
 
     def getMentions(text: String): Seq[Mention] = {
       val parsingResults = vp.parse(text)
-      parsingResults.targetMentions
+      parsingResults.allMentions
     }
 
     def test(index: Int): Unit = {
       it should s"process $index-$name correctly" in {
         if (index == -1)
           println("Put a breakpoint here to observe a particular test.")
 
-        val mentions = getMentions(text).filter(_.label matches "Assignment")
+        val allMentions = getMentions(text)
+        val mentions = allMentions.filter(_.label matches label).sortBy(_.tokenInterval)
 
-        mentions should have size variables.length
+        if (variables.isEmpty) {
+          // get only relations and events
+          val nonTextBoundMentions = mentions.filter(m => !m.isInstanceOf[TextBoundMention])
```

Comment:
Some places have a rule that this has to be .filterNot.  (It's on a best practices list somewhere.)